R/utils-get_code_dependency.R

```diff
@@ -144,6 +145,16 @@ fix_shifted_comments <- function(calls) {
   Filter(nrow, calls)
 }
 
+#' Fixes edge case of `:=` assignment operator being treated as assignment.
+#' @keywords internal
+#' @noRd
+remove_dt_assign <- function(calls) {
+  checkmate::assert_list(calls)
+  lapply(calls, function(call) {
+    call <- subset(call, !(token == "LEFT_ASSIGN" & text == ":="))
```

Comment:
What do you think about expanding the scope a tiny bit and be able to exclude more of them in the future?

```suggestion
remove_custom_assign <- function(calls, exclude = c(":=")) {
  checkmate::assert_list(calls)
  checkmate::assert_character(exclude)
  lapply(calls, function(call) {
    call[!(call$token == "LEFT_ASSIGN" & call$text %in% exclude), ]
```

Reply:
And documentation above accordingly